plugins/com.google.cloud.tools.eclipse.test.util/src/com/google/cloud/tools/eclipse/test/util/project/ProjectUtils.java

```diff
@@ -184,9 +187,12 @@ public void run() {
   /** Wait for any spawned jobs and builds to complete (e.g., validation jobs). */
   public static void waitForProjects(Runnable delayTactic, IProject... projects) {
```

Comment:
This method is pretty long, I suggest to break out parts into helper methods:
 - collecting jobs (including filtering on line 214-219
 - logic when more than 60 seconds has passed